ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -345,6 +362,68 @@ public static BlobProperties buildBlobProperties(Map<String, Object> args)
   }
 
   /**
+   * Build a {@link ByteRange} given the arguments associated with a request.
+   * @param args the arguments associated with the request.
+   * @return The {@link ByteRange} parsed from the arguments, or {@code null}, if no range argument was found
+   * @throws RestServiceException if the range could not be parsed from the header value, or it was otherwise invalid.
+   */
+  public static ByteRange buildByteRange(Map<String, Object> args)
+      throws RestServiceException {
+    String rangeValue = getHeader(args, Headers.RANGE, false);
+    if (rangeValue == null) {
+      return null;
+    }
+    if (!rangeValue.startsWith(BYTE_RANGE_PREFIX)) {
+      throw new RestServiceException("Invalid byte range syntax; does not start with 'bytes='",
+          RestServiceErrorCode.RangeNotSatisfiable);
+    }
+    try {
+      int hyphenIndex = rangeValue.indexOf('-', BYTE_RANGE_PREFIX.length());
+      String startOffsetStr = rangeValue.substring(BYTE_RANGE_PREFIX.length(), hyphenIndex);
+      String endOffsetStr = rangeValue.substring(hyphenIndex + 1);
+      if (startOffsetStr.isEmpty()) {
+        return ByteRange.fromLastNBytes(Long.parseLong(endOffsetStr));
+      } else if (endOffsetStr.isEmpty()) {
+        return ByteRange.fromStartOffset(Long.parseLong(startOffsetStr));
+      } else {
+        return ByteRange.fromOffsetRange(Long.parseLong(startOffsetStr), Long.parseLong(endOffsetStr));
+      }
+    } catch (Exception e) {
+      throw new RestServiceException("Invalid byte range syntax", e, RestServiceErrorCode.RangeNotSatisfiable);
+    }
+  }
+
+  /**
+   * Build a {@link GetBlobOptions} object from an argument map for a certain sub-resource.
+   * @param args the arguments associated with the request.
+   * @param subResource the {@link SubResource} for the request, or {@code null} if no sub-resource is requested.
+   * @return a populated {@link GetBlobOptions} object.
+   * @throws RestServiceException if the {@link GetBlobOptions} could not be constructed.
+   */
+  public static GetBlobOptions buildGetBlobOptions(Map<String, Object> args, SubResource subResource)
+      throws RestServiceException {
```

Comment:
Do you think this method could be cleaned up a bit? The check for RANGE header is done here as well as in `buildByteRange()`. I think it would be more appropriate if the check for the RANGE header is done only here and `buildByteRange()` is only called if that passes.